src/app/booking/OOSPlannerSection.js

```diff
@@ -44,14 +44,48 @@ const BookingBox = styled(Box)(({ theme }) => ({
   padding: theme.spacing(2),
 }));
 
-function FilledBooking({ booking }) {
+function useAwaitingBookingDeletion({ bookingIdInitial }) {
+  const { enqueueSnackbar } = useSnackbar();
+  const [requestNum, setRequestNum] = useState(null);
   const queryClient = useQueryClient();
+  const MAX_REQUEST_NUM = 4;
+  const isEnabled = requestNum !== null && requestNum <= MAX_REQUEST_NUM;
+
+  useQuery("deletedBooking", () => getBooking(bookingIdInitial), 
+    {
+      enabled: isEnabled,
+      refetchInterval: 2000,
+      retry: false,
+      onError: (error) => {
+        if (error.response?.status === 404) {
+          enqueueSnackbar("Booking deleted", { variant: "success" });
+          queryClient.invalidateQueries("bookings");
+        }
+        else {
+          const errorMsg = getFlatErrors(error).join("; ");
+          enqueueSnackbar(errorMsg, { variant: "error" });
+        }
+      },
+      onSuccess: () => {
+        if (requestNum === MAX_REQUEST_NUM) {
+          enqueueSnackbar("Failed to delete booking within the expected time", { variant: "error" });
+        }
+        setRequestNum(prevIterationIndex => prevIterationIndex + 1);
```

Comment:
prevIterationIndex is an old naming. You should think about consistency in naming the same way as in tech approach.